### PR TITLE
Add namespace selection

### PR DESCRIPTION
**Pull request by steveodonovan**
@a-roberts Could I get a review please ? Thanks

This is provides a basic api to get the namespaces for a cluster for selection in the UI 

https://github.com/tektoncd/dashboard/issues/86

### CODE REVIEW

File: pkg/endpoints/cluster.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	"github.com/tektoncd/dashboard/pkg/logging"
+	"github.com/tektoncd/dashboard/pkg/utils"
+	"net/http"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r Resource) getAllNamespaces(request *restful.Request, response *restful.Response) {
+	namespaces, err := r.K8sClient.CoreV1().Namespaces().List(metav1.ListOptions{});
+	logging.Log.Debugf("In getAllNamespaces - : %s", namespaces)
```
Comment (a-roberts) [inline pkg/endpoints/cluster.go]: I'd have this just be "In getAllNamespaces" in the Debugf

File: pkg/endpoints/cluster.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+	if err != nil {
+		utils.RespondError(response, err, http.StatusNotFound)
+		return
+	}
```
Comment (a-roberts) [inline pkg/endpoints/cluster.go]: Then logging.Log.Debugf("Found namespaces: %s, namespaces") here

File: pkg/endpoints/cluster_test.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import (
+	"encoding/json"
+	"net/http/httptest"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestGetAllNamespaces(t *testing.T) {
```
Comment (a-roberts) [inline pkg/endpoints/cluster_test.go]: Can you add another short test please for a namespace that doesn't exist then we test that we return the 404?

File: pkg/endpoints/router.go [hunk 4]
```diff
@@ -52,8 +52,9 @@ func (r Resource) RegisterEndpoints(container *restful.Container) {
 		Consumes(restful.MIME_JSON).
 		Produces(restful.MIME_JSON)
 
-	logging.Log.Info("Adding v1, and API for pipelines")
+	logging.Log.Info("Adding v1, and API for k8s resources and pipelines")
 
+	wsv1.Route(wsv1.GET("/").To(r.getAllNamespaces))
```
Comment (a-roberts) [inline pkg/endpoints/router.go]: Document this new endpoint under DEVELOPMENT.md please 😄 I know it's not really going to be used much by end users but at least we'll all know it's exist and what its contract is

File: pkg/endpoints/cluster_test.go (hunk 3 above)
Comment (dibbles) [reply pkg/endpoints/cluster_test.go]: I don't think this is needed @a-roberts, the namesapce is added so there is a namespace to be returned, this is not adding testing getting a specific namespace ... rather getting all namespaces.

Comment (a-roberts) [reply pkg/endpoints/cluster_test.go]: Yep you're right, my mistake 😅 will resolve this convo...

**Comment by steveodonovan** [comment]
Thanks, @a-roberts could I get another once over please ?

**Comment by a-roberts** [comment]
Looks perfect to me, thank you @steveodonovan!

/approve
/lgtm

**Pull request merged**